src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1628,6 +1634,34 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
     return false;
 }
 
+bool OutputStateMachineEngine::_ParseDirectory(const std::wstring_view string,
+                                               std::wstring& uri) const
+{
+    if (string.length() <= 8)
+    {
+        return false;
+    }
+
+    const auto prefix = string.substr(0, 7);
+    if (!prefix.compare(L"file://") == 0)
+    {
+        return false;
+    }
+
+    size_t current = 7;
+    const auto nextSlash = string.find(L"/", current);
+    if (nextSlash == std::wstring::npos)
+    {
+        return false;
+    }
+
+    // Hostname is ignored.
+    current = nextSlash + 1;
+    uri = std::wstring(string.substr(current, std::wstring::npos));

```

Comment:
This is just an early prototype. The actual implementation would require serious URL decoding and related things.